### PR TITLE
Build: Limit the numbers of cores used by the build

### PR DESCRIPTION
**Pull request by blowery**
Limit to half of the available cores. This seems to give the best performance across machines.

### CODE REVIEW

**Comment by blowery** [comment]
Prior art #25456

**Comment by blowery** [comment]
On my MacBook Pro, Core i7 (4 cores, 8 threads), no babel cache present. 
Time to build modules:

Workers | Time
---------|----------
1 | 80s 
2 | 55s 
3 | 44s
4 | 42s 
6 | 43s
8 | 45s

File: webpack.config.js [hunk 1]
```diff
@@ -72,8 +72,6 @@ const babelLoader = {
 	},
 };
 
-threadLoader.warmup( {}, [ 'babel-loader' ] );
```
[the remark this answers is not in the document]
Comment (blowery) [reply webpack.config.js]: it was an attempt to warmup the workers before they would be used. However, it has no apparent speed impact and can make the build hang, waiting for the workers to terminate.

File: webpack.config.js [hunk 2]
```diff
@@ -125,7 +123,15 @@ const webpackConfig = {
 			{
 				test: /\.jsx?$/,
 				exclude: /node_modules[\/\\](?!notifications-panel)/,
-				use: [ 'thread-loader', babelLoader ],
+				use: [
+					{
+						loader: 'thread-loader',
+						options: {
+							workers: Math.max( Math.floor( os.cpus().length / 2 ), 1 ),
```
[the remark this answers is not in the document]
Comment (blowery) [reply webpack.config.js]: Sure. I was thinking about conditionally ditching the thread-loader based on the number of cores, but defaulting to 2 is fine too.

File: webpack.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (blowery) [reply webpack.config.js]: see docs on prewarming at https://github.com/webpack-contrib/thread-loader

[the remark this answers is not in the document]
Comment (blowery) [reply webpack.config.js]: to repro the hang, add back the prewarm statement and the thread loader require, then configure the thread loader in the server build to use 4 workers. For me, `npm run build-server` would run, apparently complete, but then hang waiting for something complete.

Removing the warmup killed the hang.

**Review by DanReyLop — APPROVED** [review]
I didn't time it, but these commands still work:
* `npm start`
* `CALYPSO_ENV=production npm run build`

:shipit:

**Review by sirreal — APPROVED** [review]

**Pull request merged**